src/combinatorics/simple_graph/hasse.lean

```diff
@@ -19,6 +19,130 @@ path graph on `n` vertices.
 * `simple_graph.path_graph`: Path graph on `n` vertices.
 -/
 
+
+namespace prod
+variables {α β : Type*} [partial_order α] [partial_order β] {a a₁ a₂ : α}
+  {b b₁ b₂ : β} {x y : α × β}
+
+@[simp] lemma swap_le_swap_iff : x.swap ≤ y.swap ↔ x ≤ y := and_comm _ _
+
+@[simp] lemma swap_lt_swap_iff : x.swap < y.swap ↔ x < y :=
+lt_iff_lt_of_le_iff_le' swap_le_swap_iff swap_le_swap_iff
+
+@[simp] lemma swap_wcovby_swap_iff : x.swap ⩿ y.swap ↔ x ⩿ y :=
+apply_wcovby_apply_iff (order_iso.prod_comm : α × β ≃o β × α)
+
+@[simp] lemma swap_covby_swap_iff : x.swap ⋖ y.swap ↔ x ⋖ y :=
+apply_covby_apply_iff (order_iso.prod_comm : α × β ≃o β × α)
+
+lemma mk_le_mk_iff_left : (a₁, b) ≤ (a₂, b) ↔ a₁ ≤ a₂ := and_iff_left le_rfl
```

Comment:
I don't know if this was handled in the other PR, but this would make a good `simp` lemma, as would the three theorems below.

Reply:
Yeah all those went through in #14977.